Switch to new groups UI.

sahil839 opened the pull request:
<!-- Describe your pull request here.-->
- First commit is to add link to new groups UI in gear menu and from the group popover.
- Second commit is to add "manage group" option in popover for guests.
- Third commit is to fix the popover.
- Fourth commit is to fix the documentation. This still needs some work.
- Fifth commit is to remove old groups UI code.

Fixes: <!-- Issue link, or clear description.--> #28012

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![Screenshot from 2023-12-06 23-14-22](https://github.com/zulip/zulip/assets/35494118/058080b2-da84-4ed4-862b-1750cbb709f8)
<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

sahil839 commented:
User group popovers -

Before -
![group-popover-before](https://github.com/zulip/zulip/assets/35494118/f889ef02-42e1-4522-9a12-4afcb0276921)


After -
![group-popover-now](https://github.com/zulip/zulip/assets/35494118/9e32323a-5b0f-4775-b3c4-c7ec58ded93d)

sahil839 commented:
The help documentation changes are still incomplete. I have added helpers which can be used there but the content still needs to be updated.

alya commented:
> I am not sure how we want to handle deactivated users in both members list in groups UI and the groups popover. As seen in screenshot above, we do not show presence circle in the popover for deactivated users.

The right solution for that presence circle is a slashed circle, as in #26833.

alya commented:
@drrosa Perhaps you could build on this PR to finish the documentation updates?

alya commented:
Looks good to me otherwise!

timabbott commented:
Code change lgtm too, I can merge once those little tweaks are done.

sahil839 commented on web/src/user_group_edit_members.js:
```diff
@@ -36,9 +36,15 @@ function get_potential_members() {
-export function update_member_list_widget(group_id, member_ids) {
-    assert(group_id === current_group_id, "Unexpected group rerendering members list");
-    const users = people.get_users_from_ids(member_ids);
+function get_user_group_members(group) {
```
This function should ideally be in `user_groups.ts`, but there were some lint issues due to two `User` types. I thought we can do this as a follow-up.

timabbott replied:
OK

sahil839 commented:
Updated the PR -
- Added commits to not include deactivated users in members list. Also added commit to ignore the deactivated users when selecting user group as a DM recipient.
- Updated the popover as discussed above.
![Screenshot from 2023-12-07 11-59-33](https://github.com/zulip/zulip/assets/35494118/1de05fe6-758d-4983-bf85-e00108d2d29b)

alya commented:
Looking at it on the PR, the alignment looks off for the "Group settings" link.

1. The text is too tight to the icon. I think it would look best horizontally aligned with the names.
2. The icon looks too low next to the text. (It looks great in the gear menu.)

alya commented:
I just got an error when trying to join a group (hamletcharacters, as Desdemona):

![Screenshot 2023-12-06 at 11 52 21 PM](https://github.com/zulip/zulip/assets/2090066/ac870532-acd4-442c-8b0e-53d756102f55)

timabbott commented:
Merged, thanks @sahil839!
